src/main/java/apoc/text/Strings.java

```diff
@@ -205,4 +209,72 @@ public String slug(@Name("text") String text, @Name(value = "delim", defaultValu
         if (delim == null) return null;
         return text.trim().replaceAll("[\\W\\s]+", delim);
     }
+
+    private static final String lower = "abcdefghijklmnopqrstuvwxyz";
+    private static final String upper = lower.toUpperCase();
+    private static final String numeric = "0123456789";
+
+    @UserFunction
+    @Description("apoc.text.random(length, valid) YIELD value - generate a random string")
+    public String random(final @Name("length") long length, @Name(value = "valid", defaultValue = "A-Za-z0-9") String valid) {
+        valid = valid.replaceAll("A-Z", upper).replaceAll("a-z", lower).replaceAll("0-9", numeric);
+
+        SecureRandom rnd = new SecureRandom();
+        StringBuilder output = new StringBuilder( toIntExact(length) );
+
+        while ( output.length() < length ) {
+            output.append( valid.charAt( rnd.nextInt(valid.length()) ) );
+        }
+
+        return output.toString();
+    }
+
+    @UserFunction
+    @Description("apoc.text.capitalize(text) YIELD value - capitalise the first letter of the word")
+    public String capitalize(@Name("text") String text) {
+        return text.substring(0, 1).toUpperCase() + text.substring(1);
+    }
+
+    @UserFunction
+    @Description("apoc.text.capitalizeAll(text) YIELD value - capitalise the first letter of every word in the text")
+    public String capitalizeAll(@Name("text") String text) {
+        return WordUtils.capitalizeFully(text);
+    }
+
+    @UserFunction
+    @Description("apoc.text.decapitalize(text) YIELD value - decapitalize the first letter of the word")
+    public String decapitalize(@Name("text") String text) {
+        return StringUtils.uncapitalize(text);
+    }
+
+    @UserFunction
+    @Description("apoc.text.swapCase(text) YIELD value - Swap the case of a string")
+    public String swapCase(@Name("text") String text) {
+        return StringUtils.swapCase(text);
+    }
+
+    @UserFunction
+    @Description("apoc.text.camelCase(text) YIELD value - Convert a string to camelCase")
+    public String camelCase(@Name("text") String text) {
+        String converted = WordUtils.capitalizeFully(text);
```

Comment:
I don't think this works.

it would also be good to turn e.g. "FOO_BAR" or "foo-bar" or "FooBar" into "fooBar"